x-pack/plugins/ml/public/components/job_selector/job_select_service_utils.js

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { difference, find } from 'lodash'; // TODO: find a way to not rely on this anymore
+import { toastNotifications } from 'ui/notify';
+import { mlJobService } from '../../services/job_service';
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+import d3 from 'd3';
+
+
+function warnAboutInvalidJobIds(invalidIds) {
+  if (invalidIds.length > 0) {
+    toastNotifications.addWarning(i18n.translate('xpack.ml.jobSelect.requestedJobsDoesNotExistWarningMessage', {
+      defaultMessage: `Requested
+{invalidIdsLength, plural, one {job {invalidIds} does not exist} other {jobs {invalidIds} do not exist}}`,
+      values: {
+        invalidIdsLength: invalidIds.length,
+        invalidIds,
+      }
+    }));
+  }
+}
+
+// check that the ids read from the url exist by comparing them to the
+// jobs loaded via mlJobsService.
+function getInvalidJobIds(ids) {
+  return ids.filter(id => {
+    const job = find(mlJobService.jobs, { 'job_id': id });
+    return (job === undefined && id !== '*');
```

Comment:
If I get the purpose of this function right this could be something like this without lodash:
```js
const jobExists = mlJobService.jobs.includes(job => job.job_id === id);
return (jobExists === false && id !== '*');
```

Reply:
I don't think includes takes a function but we can definitely use `some` for the `jobExists` bit and remove the use of `find` 👍 Nice catch! Thanks 😄